compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2818,6 +2803,56 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr, B
         return actualType;
     }
 
+    private BType checkTupleIndexBasedAccess(BLangIndexBasedAccess accessExpr, BType indexExprType, BTupleType tuple) {
+        BType actualType = symTable.semanticError;
+        BLangExpression indexExpr = accessExpr.indexExpr;
+        switch (indexExprType.tag) {
```

Comment:
If we remove the `indexExprType` parameter(), we can use `indexExpr` to get the type tag here. WDYT?